src/cli/commands/install.js

```diff
@@ -282,7 +282,12 @@ export class Install {
         }
       }
 
-      const pushDeps = (depType, manifest: Object, {hint, optional}, isUsed) => {
+      const pushDeps = (
+        depType,
+        manifest: Object,
+        {hint, optional}: {hint: ?constants.RequestHint, optional: boolean},
```

Comment:
Why make `hint` optional here?

Reply:
Because of https://github.com/yarnpkg/yarn/blob/5ef113f7db1201f48e22498374023ebe1e61b069/src/cli/commands/install.js#L327

Not sure if `constants.RequestHint | null` or `?constants.RequestHint` is preferred(?) Or maybe add a `production` hint and always require `hint`?